src/imap.rs

```diff
@@ -621,11 +618,18 @@ impl Imap {
                 .await
                 .with_context(|| format!("STATUS (UIDNEXT) error for {folder:?}"))?;
 
-            status
-                .uid_next
-                .with_context(|| format!("STATUS {folder} (UIDNEXT) did not return UIDNEXT"))?
+            if status.uid_next.is_none() {
+                // This happens with mail.163.com as of 2023-11-26.
+                // It does not return UIDNEXT on SELECT and returns invalid
+                // `* STATUS "INBOX" ()` response on explicit request for UIDNEXT.
+                warn!(context, "STATUS {folder} (UIDNEXT) did not return UIDNEXT.");
+            }
+            status.uid_next
         };
-        mailbox.uid_next = Some(new_uid_next);
+        if new_uid_next.is_some() {
+            mailbox.uid_next = new_uid_next;
+        }
+        let new_uid_next = new_uid_next.unwrap_or_default();
```

Comment:
Doesn't it need to be set to `old_uid_next` by default? Otherwise we will reset the folder's uid_next to 0 further with the warning "The server illegally decreased the uid_next of folder..."